docs/TAP_SUBMISSION_REQUIREMENTS.md

```diff
@@ -0,0 +1,161 @@
+# TAP SUBMISSION REQUIREMENTS
+This is the doc that will outline all requirements for tap contributors.
+
+Before working on your project, please check that someone else has not submitted the same project already.
+The [Singer website][singer-io] and the [Singer Slack][singer-slack] are great resources for this.
+
+
+Before submitting your tap for review, make sure that you have completed all of the following:
+
+- [Project Structure](#projectstructure)
+- [Catalog](#catalog)
+- [Stream selection](#stream-selection)
+- [Field selection](#field-selection)
+- [Child Streams Handled Correctly](#how-to-handle-child-streams)
+- [Pylint](#pylint)
+- [PyPi](#pypi)
+- [Contributor License Agreement](#contributor-license-agreement)
+
+## Project Structure
+
+The project should have the following structure:
+
+```
+tap-<tap_name>
+├── LICENSE
+├── MANIFEST.in
+├── README.md
+├── setup.py
+├── tap_<tap_name>
+│   ├── schemas
+│   │   ├── <stream_name>.json
+│   │   ├── <stream_name>.json
+│   │   └── <stream_name>.json
+│   ├── client.py
+│   ├── discover.py
+│   ├── __init__.py
+│   ├── streams.py
+│   └── sync.py
+└── tests
+    └── base.py
+```
+files/directories to note:
+- `schemas`: This folder houses all the json schema files. Each stream has a json schema associated with it, as defined in this file.
+- `client.py`: Calls to the api are made here.
+- `discover.py`: Discovery code is here.
+- `__init__.py`: The main function is in this file. Grab command line arguments and kick off discovery/sync here.
+- `streams.py`: If using the class-based model, this is where all the stream classes and their corresponding functions live.
+- `sync.py`: This is where the sync function is run
+
+
+## Catalog
+In order for sync to run properly, the tap must correctly output a catalog during discovery.
+
+The catalog structure is described [here][catalog]
+
+An easy way to construct the catalog is by using the following call:
+
+```
+catalog_entry = {
+    "stream": stream_name,
+    "tap_stream_id": stream_name,
+    "schema": schema,
+    "metadata": metadata.get_standard_metadata(
+        schema=schema,
+        key_properties=stream.key_properties,
+        valid_replication_keys=stream.replication_keys,
+        replication_method=stream.replication_method,
+    ),
+    "key_properties": stream.key_properties
+}
+```
+(see [here][adroll-discovery] for the full example)
```

Comment:
I would elaborate the link visible text from `here` to `tap-adroll's implementation`. I would also capitalize the first word. Maybe just make it a sentence instead of putting it in parentheses. Or italicize if if you prefer to differentiate it from the other text more...